cni/pkg/ebpf/server/redirectionServer.go

```diff
@@ -0,0 +1,542 @@
+		}
+	} else {
+		// ip slice is just in network endian
+		ip := net.ParseIP(ipAddr).To4()
+		if ip == nil {
+			return fmt.Errorf("invalid ip addr(%s), ipv4 is supported", ipAddr)
+		}
+		if remove {
+			if err := r.detachTCForWorkload(uint32(ifindex)); err != nil {
+				multiErr = concatError(multiErr, err)
+			}
+			if err := r.obj.AppInfo.Delete(ip); err != nil {
+				multiErr = concatError(multiErr, err)
+			}
+		} else {
+			if err := r.attachTCForWorkLoad(uint32(ifindex)); err != nil {
+				multiErr = concatError(multiErr, err)
+			}
+			if err := r.obj.AppInfo.Update(ip, mapInfo, ebpf.UpdateAny); err != nil {
+				multiErr = concatError(multiErr, err)
+			}
+		}
+	}
+	return multiErr
+}
+
+// temp usage, will remove when integration with CNI control plane
+func (r *RedirectServer) Redirect(w http.ResponseWriter, req *http.Request) {
+	fmt.Println("Get new post request")
+	ifindex, _ := strconv.Atoi(req.URL.Query().Get("i"))
+	peerIndex := 2
+	namespace := req.URL.Query().Get("n")
+	ztunnel, _ := strconv.ParseBool(req.URL.Query().Get("z"))
+	remove, _ := strconv.ParseBool(req.URL.Query().Get("r"))
+	ip := req.URL.Query().Get("p")
+	mac := req.URL.Query().Get("m")
+	args := RedirectArgs{ifindex: ifindex, peerIndex: peerIndex, ztunnel: ztunnel, namespace: namespace, remove: remove, macAddr: mac, ipAddr: ip}
+	r.AcceptRequest(&args)
+	fmt.Fprintf(w, "Applied %+v\n", args)
+}
+
+type ztunnelInfo struct {
+	Ifindex uint32 `json:"ifindex"`
+	MacAddr string `json:"mac_addr"`
+}
+
+// temp usage, will remove when integration with CNI control plane
+//
+//nolint:errcheck
+func (r *RedirectServer) DumpZtunnelInfo(w http.ResponseWriter, req *http.Request) {
+	v, err := r.dumpZtunnelInfo()
+	if err != nil {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		w.Write([]byte(err.Error()))
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	var info ztunnelInfo
+	info.Ifindex = v.Ifindex
+	info.MacAddr = net.HardwareAddr(v.MacAddr[:]).String()
+	bytes, _ := json.Marshal(info)
+	w.Write(bytes)
+}
+
+type appInfo struct {
+	IPAddr  string `json:"ipaddr"`
+	Ifindex uint32 `json:"ifindex"`
+	MacAddr string `json:"mac_addr"`
+}
+
+// temp usage, will remove when integration with CNI control plane
+//
+//nolint:errcheck
+func (r *RedirectServer) DumpAppInfo(w http.ResponseWriter, req *http.Request) {
+	k, v := r.dumpAppInfo()
+	if len(k) != len(v) {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	info := make([]appInfo, len(k))
+	for i := range k {
+		ip := htonl(k[i])
+		info[i].IPAddr = net.IPv4(byte(ip>>24), byte(ip>>16), byte(ip>>8), byte(ip)).String()
+		info[i].Ifindex = v[i].Ifindex
+		info[i].MacAddr = net.HardwareAddr(v[i].MacAddr[:]).String()
+	}
+	bytes, _ := json.Marshal(info)
+	w.Write(bytes)
+}
+
+func (r *RedirectServer) AcceptRequest(redirectArgs *RedirectArgs) {
+	r.redirectArgsChan <- redirectArgs
+}
+
+func (r *RedirectServer) attachTCForZtunnel(ifindex, peerIndex uint32, namespace string) error {
```

Comment:
What happens during the time between when L328 and L332 execute? And we have only attached to one part of the veth?